showcase/app/components/page-components/copy/button/code-fragments/with-masked-input.gts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { TemplateOnlyComponent } from '@ember/component/template-only';
+import { eq } from 'ember-truth-helpers';
+
+import {
+  HdsCopyButton,
+  HdsFormMaskedInputBase,
+  HdsFormField,
+  HdsFormMaskedInputField,
+} from '@hashicorp/design-system-components/components';
+
+type variant =
+  | 'masked-input-base'
+  | 'masked-input-base-form-field'
+  | 'masked-input-field';
+
+export const MASKED_INPUT_VARIANTS: variant[] = [
+  'masked-input-base',
+  'masked-input-base-form-field',
+  'masked-input-field',
+];
+
+export interface CodeFragmentWithMaskedInputSignature {
+  Args: {
+    isMultiline?: boolean;
+    variant: variant;
+  };
+  Element: HTMLDivElement;
+}
+
+const CodeFragmentWithMaskedInput: TemplateOnlyComponent<CodeFragmentWithMaskedInputSignature> =
+  <template>
+    <div ...attributes>
```

Comment:
[personal consideration] we're adding `...attributes` even if not needed in the instantiation. what do you think of taking a more minimalistic approach, and add them only when needed?